keps/sig-scheduling/3280-guarantee-pdb-when-preemption-happens/README.md

```diff
@@ -0,0 +1,977 @@
+allowDisruptionByPriorityGreaterThanOrEqual: 1000
+```
+
+The scheduler plugin `defaultpreemption` needs to check the value set in the `AllowDisruptionByPriorityGreaterThanOrEqual` field 
+when selecting victims.
+- if the priority of the preemptor is greater than or equal to the value of `AllowDisruptionByPriorityGreaterThanOrEqual` in victim pod, 
+  the implementation will remain consistent with the existing behavior, meaning that the scheduler will try to select victims whose PDBs 
+  are not violated by preemption, but if no such victims are found, preemption will still happen and lower priority pods will be removed 
+  despite their PDBs being violated.
+- if the priority of the preemptor is less than the value of `AllowDisruptionByPriorityGreaterThanOrEqual` in victim pod, 
+  the scheduler will check if the victim' PDBs will be violated when selecting victims
+  - if violate the victims' PDBs, this victim will not be selected as candidates.
+  - if not violate the victims' PDBs, scheduler will preempt this pod via the `/evictions` endpoint. If it responds `200 OK`, 
+    it means the eviction is allowed, and the victim is deleted, similar to sending a DELETE request to the Pod URL. If it responses
+    `429 Too Many Requests`, the scheduler will output an error log and choose another victim among the candidate victims to preempt 
+    until it succeeds or there are no more candidates.
+
+### PreemptionPolicy vs AllowDisruptionByPriorityGreaterThanOrEqual
+
+The `PreemptionPolicy` is used to describe the behavior of the pods associated with the PriorityClass during preemption as the preemptor. 
+The `AllowDisruptionByPriorityGreaterThanOrEqual` is used to describe the policy of the pods associated with the PriorityClass during 
+preemption as the victim. During a preemption process, if the preemptor is configured with the `PreemptionPolicy` and the victim is 
+configured with the `AllowDisruptionByPriorityGreaterThanOrEqual`, the `AllowDisruptionByPriorityGreaterThanOrEqual` takes priority over 
+the `PreemptionPolicy`.
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+
+##### Unit tests
```

Comment:
Didn't read the codes in priorityClass, don't we need a bit of tests there? Or I just missed in the KEP.

Reply:
The test codes relevant PriorityClass is in these pkgs: `pkg/scheduler/apis/config/v1` and `k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption`. I will add the additional tests in these pkgs.